### PR TITLE
actually use the new memory intrinsics

### PR DESCRIPTION
**Pull request by andrewrk**
Follow-up from #15278. This branch updates the codebase to actually switch from `std.mem.set` and `std.mem.copy` to `@memset` and `@memcpy` respectively.

I had to update zig1.wasm again because of bugs that this uncovered in the C backend.

There are a few more instances to update before merging.

### CODE REVIEW

**Comment by andrewrk** [comment]
So far, with the crypto benchmark script I'm seeing about a 10% slowdown in the throughput of most crypto functions (which make heavy use of memcpy).

This is interesting, because it's the opposite of what result I would expect to see based on these changes. I'll take a look at the machine code in the hot loops of one of the algorithms and see what is happening.

Possibly related: #15482

File: src/codegen/c.zig [hunk 1]
```diff
@@ -6288,28 +6334,42 @@ fn airMemset(f: *Function, inst: Air.Inst.Index, safety: bool) !CValue {
...
+            try writer.writeAll("((");
+            try f.renderType(writer, elem_ptr_ty);
+            try writer.writeByte(')');
+            try writeSliceOrPtr(f, writer, dest_slice, dest_ty);
+            try writer.writeAll(")[");
+            try f.writeCValue(writer, index, .Other);
+            try writer.writeAll("] = ");
+            try f.writeCValue(writer, value, .FunctionArgument);
+            try writer.writeAll(";\n");
+        }
```
Comment (jacobly0) [inline src/codegen/c.zig:6357]:
```suggestion
        const a = try Assignment.start(f, writer, elem_ty);
        try writer.writeAll("((");
        try f.renderType(writer, elem_ptr_ty);
        try writer.writeByte(')');
        try writeSliceOrPtr(f, writer, dest_slice, dest_ty);
        try writer.writeAll(")[");
        try f.writeCValue(writer, index, .Other);
        try writer.writeByte(']');
        try a.assign(f, writer);
        try f.writeCValue(writer, value, .Other);
        try a.end(f, writer);
```

Comment (jacobly0) [reply src/codegen/c.zig:6357]: I recently factored out these maybe-`memcpy`s.

**Review by jacobly0 — CHANGES_REQUESTED** [review]

File: src/codegen/c.zig (hunk 1 above)
Comment (andrewrk) [reply src/codegen/c.zig:6357]: Beautiful, thank you!

**Comment by frmdstryr** [comment]
Please label as breaking or make it backwards compatible...

**Pull request merged**